### PR TITLE
fix failures of pl_device_test with shots set

### PR DESCRIPTION
**Pull request by multiphaseCFD**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

Currently there are failures when running pl_device_test with shots set with a certain number. Following the Lightning-GPU implementation, `sample()` is added to the python layer to fix this issue.

**Description of the Change:**

**Benefits:**

Avoid failures when running pl_device_test with shots set with a certain number.

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Review by AmintorDusko — APPROVED** [review]
Looks good to me! Thank you for that!

File: .github/workflows/tests_linux.yml [hunk 1]
```diff
@@ -80,7 +80,8 @@ jobs:
         run: |
           cd main/
           pytest tests/
-          pl-device-test --device lightning.kokkos
+          pl-device-test --device lightning.kokkos --skip-ops --shots=20000
+          pl-device-test --device lightning.kokkos --shots=None --skip-ops
```
Comment (vincentmr) [inline .github/workflows/tests_linux.yml:84]: Is there a reason for the reverse order in the arguments here?

**Review by vincentmr — APPROVED** [review]
Looks good, thanks!

File: .github/workflows/tests_linux.yml (hunk 1 above)
Comment (multiphaseCFD) [reply .github/workflows/tests_linux.yml:84]: Not really, just followed previous implementation.

File: pennylane_lightning_kokkos/lightning_kokkos.py [hunk 2]
```diff
@@ -287,6 +287,12 @@ def probability(self, wires=None, shot_range=None, bin_size=None):
 
         return self._kokkos_state.probs(device_wires)
 
+    def sample(self, observable, shot_range=None, bin_size=None, counts=False):
+        if observable.name != "PauliZ":
+            self.apply_kokkos(observable.diagonalizing_gates())
+            self._samples = self.generate_samples()
+        return super().sample(observable, shot_range=shot_range, bin_size=bin_size, counts=counts)
```
Comment (mlxd) [inline pennylane_lightning_kokkos/lightning_kokkos.py:294]: Great catch! This pesky diagonalizing gate issue was a tricky one to find.

**Review by mlxd — APPROVED** [review]
Nice work @multiphaseCFD 
Happy to see this go through.

**Pull request merged**